src/modules/core/components/time-field/time-field.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  type FC,
+  type ChangeEvent,
+  useCallback,
+  useRef,
+  useState,
+  useEffect,
+} from 'react';
+import { useBoolean, useUpdateEffect } from 'usehooks-ts';
+// components
+import { TextField, TimeSelect } from '@/modules/core/components';
+import { Popover } from '@/modules/core/components/popover';
+// hooks
+import { useOnClickOutsideExtended } from '@/modules/core/hooks/use-on-click-outside-extended';
+// utils
+import { Time } from '@/modules/core/utils/time.utils';
+
+import type { TimeFieldProps } from './time-field.interface';
+import styles from './time-field.module.scss';
+
+export const TimeField: FC<TimeFieldProps> = ({
+  value,
+  onChange,
+  inputProps,
+}) => {
+  // state
+  const isActive = useBoolean();
+  const [time, setTime] = useState(new Time());
+  // refs
+  const rootRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const popoverRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (Time.guard(value)) {
+      setTime(new Time(value));
+    }
+  }, [value]);
+
+  useUpdateEffect(() => {
+    onChange(time.getString());
+  }, [time]);
+
+  useOnClickOutsideExtended(
+    rootRef,
+    () => {
+      isActive.setFalse();
+    },
+    [popoverRef]
+  );
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      // Remove any non-digit characters from the input value
+      const sanitizedValue = e.target.value.replace(/\D/g, '');
+
+      // Format the sanitized value with colons
+      let formattedValue = '';
+
+      for (let i = 0; i < sanitizedValue.length; i++) {
```

Comment:
Honestly it looks as some hack to maby, maybe it's better if you put it in some helper.